File: app/src/main/java/net/ktnx/mobileledger/json/v1_40/AccountListParser.java
/*
 * Copyright © 2020 Damyan Ivanov.
 * This file is part of MoLe.
 * MoLe is free software: you can distribute it and/or modify it
 * under the term of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your opinion), any later version.
 *
 * MoLe is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License terms for details.
 *
 * You should have received a copy of the GNU General Public License
 * along with MoLe. If not, see <https://www.gnu.org/licenses/>.
 */

package net.ktnx.mobileledger.json.v1_40;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.ObjectReader;

import net.ktnx.mobileledger.json.API;

import java.io.IOException;
import java.io.InputStream;

public class AccountListParser extends net.ktnx.mobileledger.json.AccountListParser {

    public AccountListParser(InputStream input) throws IOException {
        ObjectMapper mapper = new ObjectMapper();
        ObjectReader reader = mapper.readerFor(ParsedLedgerAccount.class);

        iterator = reader.readValues(input);
    }
    @Override
    public API getApiVersion() {
        return API.v1_19_1;
    }
}

File: app/src/main/kotlin/net/ktnx/mobileledger/json/v1_19_1/AccountListParser.kt
/*
 * Copyright © 2020 Damyan Ivanov.
 * This file is part of MoLe.
 * MoLe is free software: you can distribute it and/or modify it
 * under the term of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your opinion), any later version.
 *
 * MoLe is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License terms for details.
 *
 * You should have received a copy of the GNU General Public License
 * along with MoLe. If not, see <https://www.gnu.org/licenses/>.
 */

package net.ktnx.mobileledger.json.v1_19_1

import com.fasterxml.jackson.databind.ObjectMapper
import java.io.InputStream
import net.ktnx.mobileledger.json.API
import net.ktnx.mobileledger.json.AccountListParser as BaseParser

class AccountListParser(input: InputStream) : BaseParser() {
    override val apiVersion: API = API.v1_19_1

    init {
        val mapper = ObjectMapper()
        val reader = mapper.readerFor(ParsedLedgerAccount::class.java)
        iterator = reader.readValues(input)
    }
}
